bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -115,4 +129,70 @@
      */
     public void runNow(String ruleUID);
 
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a given scope or {@code null} scope.
+     * @param scope to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasScope(String scope) {
```

Comment:
I can extract these to a dedicated class although IHMO they make the common `Predicate`s for `Rule` search readily available and because of their static nature they don't put any burden on the developer who wants to implement his own `RuleRegistry`...